PWGHF/TableProducer/candidateSelectorDstar.cxx

```diff
@@ -0,0 +1,348 @@
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::aod;
+using namespace o2::analysis;
+
+// Struct to applying Dstar selection cuts
+struct HfCandidateSelctorDstar {
+  Produces<aod::HfSelDstarToD0Pi> hfSelDstarCandidate;
+
+  // Configurable specific to D0
+  Configurable<double> ptD0CandMin{"ptD0CandMin", 0., "Lower bound of D0 candidate pT"};
+  Configurable<double> ptD0CandMax{"ptD0CandMax", 50., "Upper bound of D0 candidate pT"};
+  Configurable<std::vector<double>> binsPtForD0{"binsPtForD0", std::vector<double>{hf_cuts_d0_to_pi_k::vecBinsPt}, "pT bin limits for D0"};
+  Configurable<LabeledArray<double>> cutsForD0{"cutsForD0", {hf_cuts_d0_to_pi_k::cuts[0], hf_cuts_d0_to_pi_k::nBinsPt, hf_cuts_d0_to_pi_k::nCutVars, hf_cuts_d0_to_pi_k::labelsPt, hf_cuts_d0_to_pi_k::labelsCutVar}, "D0 candidate selection per pT bin"};
+
+  // Configurable specific to Dstar
+  Configurable<double> ptDstarCandMin{"ptDstarCandMin", 0., "Lower bound of Dstar candidate pT"};
+  Configurable<double> ptDstarCandmax{"ptDstarCandmax", 50., "upper bound of Dstar candidate pT"};
+  Configurable<std::vector<double>> binsPtForDstar{"binsPtForDstar", std::vector<double>{hf_cuts_dstar_to_pi_d0::vecBinsPt}, "pT bin limits for Dstar"};
+  Configurable<LabeledArray<double>> cutsForDstar{"cutsForDstar", {hf_cuts_dstar_to_pi_d0::cuts[0], hf_cuts_dstar_to_pi_d0::nBinsPt, hf_cuts_dstar_to_pi_d0::nCutVars, hf_cuts_dstar_to_pi_d0::labelsPt, hf_cuts_dstar_to_pi_d0::labelsCutVar}, "Dstar candidate selection per pT bin"};
+
+  // common Configurable
+  // TPC PID
+  Configurable<double> ptPidTpcMin{"ptPidTpcMin", 0.15, "Lower bound of track pT for TPC PID"};
+  Configurable<double> ptPidTpcMax{"ptPidTpcMax", 5., "Upper bound of track pT for TPC PID"};
+  Configurable<double> nSigmaTpcMax{"nSigmaTpcMax", 3., "Nsigma cut on TPC only"};
+  Configurable<double> nSigmaTpcCombinedMax{"nSigmaTpcCombinedMax", 5., "Nsigma cut on TPC combined with TOF"};
+
+  // TOF PID
+  Configurable<double> ptPidTofMin{"ptPidTofMin", 0.15, "Lower bound of track pT for TOF PID"};
+  Configurable<double> ptPidTofMax{"ptPidTofMax", 5., "Upper bound of track pT for TOF PID"};
+  Configurable<double> nSigmaTofMax{"nSigmaTofMax", 3., "Nsigma cut on TOF only"};
+  Configurable<double> nSigmaTofCombinedMax{"nSigmaTofCombinedMax", 5., "Nsigma cut on TOF combined with TPC"};
+
+  // AND logic for TOF+TPC PID
+  Configurable<bool> usePidTpcAndTof{"usePidTpcAndTof", false, "Use AND logic for TPC and TOF PID"};
+
+  // CCDB configuration
+  Configurable<std::string> ccdbUrl{"ccdbUrl", "http://alice-ccdb.cern.ch", "url of the ccdb repository"};
+
+  TrackSelectorPi selectorPion;
+  TrackSelectorKa selectorKaon;
+  HfHelper::D0FromDstar hfHelper;
+
+  using TracksSel = soa::Join<aod::TracksWDcaExtra, aod::TracksPidPi, aod::TracksPidKa>;
+  // using TracksSel = soa::Join<aod::Tracks, aod::TracksPidPi, aod::TracksPidKa>;
+
+  double massPi, massK, massD0;
+  void init(InitContext& initContext)
```

Comment:
Fix order and spacing

Reply:
ordering on what basis?

Reply:
On the basis of the PWGHF coding guidelines on the O2 analysis framework documentation page.